src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
+inline bool any_of(Container&& c, Predicate p) {
+    using std::begin; //ADL
```

Comment:
Would you suggest foregoing the comments altogether, or being pessimistic and assuming people browsing the code aren't aware of ADL? I chose this as a compromise at the time.